sm/sm-sbi.c

```diff
@@ -11,123 +12,167 @@
 #include "platform.h"
 #include "plugins/plugins.h"
 
+static spinlock_t sbi_lock = SPINLOCK_INIT;
+
+void lock_sbi() {
+  while (spinlock_trylock(&sbi_lock)) {
+    pmp_ipi_update();
+  }
+}
+
+void unlock_sbi() {
+  spinlock_unlock(&sbi_lock);
+}
+
 uintptr_t mcall_sm_create_enclave(uintptr_t create_args)
 {
+  lock_sbi();
   struct keystone_sbi_create create_args_local;
   enclave_ret_code ret;
 
   /* an enclave cannot call this SBI */
   if (cpu_is_enclave_context()) {
+    unlock_sbi();
     return ENCLAVE_SBI_PROHIBITED;
   }
 
   ret = copy_from_host((struct keystone_sbi_create*)create_args,
                        &create_args_local,
                        sizeof(struct keystone_sbi_create));
 
-  if( ret != ENCLAVE_SUCCESS )
+  if( ret != ENCLAVE_SUCCESS ) {
+    unlock_sbi();
```

Comment:
Now that we have locks here, please switch this function from a `return foo;` to a `ret_code = foo; goto error;` model.

Reply:
will do